### PR TITLE
Added unit tests for associate / disassociate

### PR DESCRIPTION
**Pull request by breillyr**

### CODE REVIEW

File: tests/test_pulp_instance.py [hunk 1]
```diff
@@ -278,6 +278,20 @@ def test_checkBlobsFail(self, pulp, repo, blob):
         response = pulp.checkBlobs(repo, blobs)
         assert response['error']
 
+    @pytest.mark.parametrize('dist_id', ['foo', 'bar'])
+    def test_associate(self, pulp, dist_id):
+        if dist_id == 'bar':
+            with pytest.raises(SystemExit):
```
Comment (twaugh) [inline tests/test_pulp_instance.py]: Why `SystemExit`? Shouldn't this be `KeyError`?

Also do we expect `RequestsHttpCaller.__call__` to be called at all for this?

File: tests/test_pulp_instance.py [hunk 2]
```diff
@@ -318,3 +332,22 @@ def test_createRepo(self, pulp, repo_id, url, registry_id, sig, distributors, pr
...
+        dist_id = 'foo'
+        dl = '/pulp/api/v2/repositories/%s/distributors/%s/' % (repo, dist_id)
+        tid = '123'
+        url = '/pulp/api/v2/tasks/%s/' % tid
+        t = {'state': 'finished'}
+        flexmock(RequestsHttpCaller)
+        (RequestsHttpCaller
+            .should_receive('__call__')
+            .with_args('delete', dl)
+            .and_return(tid))
```
Comment (twaugh) [inline tests/test_pulp_instance.py]: `.once()`?

File: tests/test_pulp_instance.py [hunk 3]
```diff
@@ -318,3 +332,22 @@ def test_createRepo(self, pulp, repo_id, url, registry_id, sig, distributors, pr
...
+        (RequestsHttpCaller
+            .should_receive('__call__')
+            .with_args('get', url)
+            .and_return(t))
```
Comment (twaugh) [inline tests/test_pulp_instance.py]: `.once()`?

File: tests/test_pulp_instance.py (hunk 1 above)
Comment (breillyr) [inline tests/test_pulp_instance.py]: The KeyError is excepted in the code so it wouldn't be what we're looking for here.
I did however decide to change the systemexit to a configuration error, to make it more clear
to the end user what the problem is.

We do not expect that call for this check - that's specifically the pulp '_delete' which is never reached if the distributor id is incorrect.

Comment (twaugh) [inline tests/test_pulp_instance.py]: OK.

Could add a
```
(flexmock(RequestsHttpCaller)
    .should_receive('__call__')
    .never())
```

**Pull request merged**